presto-client/src/main/java/com/facebook/presto/client/FixJsonDataUtils.java

```diff
@@ -201,4 +227,51 @@ private static Object fixValue(TypeSignature signature, Object value)
                 return value;
         }
     }
+
+    private static Map<String, String> fixObjectFromMap(String value)
```

Comment:
Can you please add a comment explaining why these methods are necessary?  e.g. that only one level of parsing is happening?